models/ReadActivity.js

```diff
@@ -0,0 +1,149 @@
+ *
+ * @property {Publication} publication - Returns the publication, if any, that was acted upon.
+ *
+ * `Activity` models the stored activities themselves. `Add`, `Create`, `Remove`, `Undo`, `Redo`, `Update`, and `Read` activities are all stored. `Read` is unique in that it models a _client-side_ not servers side activity. (May add `View` at a later date if needed.)
+ *
+ * The document, note, and publication fields on the `relationMappings` property are not exclusive. You can have an activity on a document, with no note or publication. And you can have an activity that is the addition of a note to a document in the context of a publication.
+ */
+class ReadActivity extends BaseModel {
+  static get tableName () /*: string */ {
+    return 'readActivity'
+  }
+
+  get path () /*: string */ {
+    return 'readActivity'
+  }
+
+  static get jsonSchema () /*: any */ {
+    return {
+      type: 'object',
+      properties: {
+        id: { type: 'string' },
+        selector: { type: 'object' },
+        json: { type: 'object' },
+        readerId: { type: 'string' },
+        publicationIn: { type: 'string' },
+        published: { type: 'string', format: 'date-time' }
+      },
+      required: ['readerId', 'publicationId']
+    }
+  }
+
+  static get relationMappings () /*: any */ {
+    const { Publication } = require('./Publication.js')
+    const { Reader } = require('./Reader.js')
+    const { Document } = require('./Document.js')
+    const { Note } = require('./Note.js')
+    const { Tag } = require('./Tag.js')
+
+    /*
+    assert.ok(Model)
+    assert.ok(Model.BelongsToOneRelation)
+    assert.ok(Reader)
+    assert.ok(Document)
+    assert.ok(Note)
+    assert.ok(Publication)
+    */
+    return {
+      reader: {
+        relation: Model.BelongsToOneRelation,
+        modelClass: Reader,
+        join: {
+          from: 'readActivity.readerId',
+          to: 'Reader.id'
+        }
+      },
+      publication: {
+        relation: Model.BelongsToOneRelation,
+        modelClass: Publication,
+        join: {
+          from: 'readActivity.publicationId',
+          to: 'Publication.id'
+        }
+      }
+    }
+  }
+
+  // static async byId (id /*: string */) /*: Promise<activity> */ {
+  //   return await ReaderActivity.query()
+  //     .findById(id)
+  //     .eager('readerActivity')
+  // }
+
+  static async createReadActivity (
+    readerId /*: string */,
+    publicationId /*: string */,
+    object /*: any */
+  ) /*: any */ {
+    if (!readerId) return new Error('wrong readerId')
+
+    if (!publicationId) return new Error('wrong publicationId')
+
+    if (!object) return new Error('wrong object')
+
+    const props = _.pick(object, ['selector', 'json'])
+
+    props.id = createId()
+    props.readerId = readerId
+    props.publicationId = publicationId
+    props.published = new Date().toISOString()
+
+    try {
+      var activity = await ReadActivity.query()
```

Comment:
to make things consistent, you should use either 'const' or 'let' instead of 'var'